plugins/main/bridge/bridge.go

```diff
@@ -211,38 +216,48 @@ func cmdAdd(args *skel.CmdArgs) error {
 		return err
 	}
 
-	// TODO: make this optional when IPv6 is supported
-	if result.IP4 == nil {
-		return errors.New("IPAM plugin returned missing IPv4 config")
+	if len(result.IP) == 0 {
+		return errors.New("IPAM plugin returned missing IP config")
 	}
+	result.Interfaces = []*types.Interface{hostInterface, containerInterface}
 
-	if result.IP4.Gateway == nil && n.IsGW {
-		result.IP4.Gateway = calcGatewayIP(&result.IP4.IP)
+	for _, ipc := range result.IP {
+		// All IPs currently refer to the container interface
+		ipc.Interface = 1
+		if ipc.Gateway == nil && n.IsGW {
+			ipc.Gateway = calcGatewayIP(&ipc.Address)
+		}
 	}
 
 	if err := netns.Do(func(_ ns.NetNS) error {
 		// set the default gateway if requested
 		if n.IsDefaultGW {
-			_, defaultNet, err := net.ParseCIDR("0.0.0.0/0")
-			if err != nil {
-				return err
-			}
+			for _, ipc := range result.IP {
+				defaultNet := &net.IPNet{
+					IP:   net.IPv6zero,
+					Mask: net.IPMask(net.IPv6zero),
+				}
+				if ipc.Address.IP.To4() != nil {
+					defaultNet.IP = net.IPv4zero
+					defaultNet.Mask = net.IPMask(net.IPv4zero)
+				}
 
-			for _, route := range result.IP4.Routes {
-				if defaultNet.String() == route.Dst.String() {
-					if route.GW != nil && !route.GW.Equal(result.IP4.Gateway) {
-						return fmt.Errorf(
-							"isDefaultGateway ineffective because IPAM sets default route via %q",
-							route.GW,
-						)
+				for _, route := range ipc.Routes {
+					if defaultNet.String() == route.Dst.String() {
+						if route.GW != nil && !route.GW.Equal(ipc.Gateway) {
+							return fmt.Errorf(
+								"isDefaultGateway ineffective because IPAM sets default route via %q",
+								route.GW,
+							)
+						}
 					}
 				}
-			}
 
-			result.IP4.Routes = append(
-				result.IP4.Routes,
-				types.Route{Dst: *defaultNet, GW: result.IP4.Gateway},
-			)
+				ipc.Routes = append(
+					ipc.Routes,
+					types.Route{Dst: *defaultNet, GW: ipc.Gateway},
```

Comment:
Do we need to add an increasing metric here to make the default route deterministic?